interfaces/apparmor/backend.go

```diff
@@ -169,12 +169,14 @@ func reloadProfiles(profiles []string) error {
 	return nil
 }
 
-func unloadProfiles(profiles []string) error {
+func unloadAndRemoveCachedProfiles(profiles []string) error {
```

Comment:
Let's please keep the original function name. The point is still to unload the profile. The fact we clean up the cache is an implementation detail of it.